### ISSUE TITLE
[4.0.0-beta.5] icon slot margins in ion-button are incorrect in RTL mode

### ISSUE THREAD

**Issue by abennouna** [ionic-team/ionic#15361]
# Bug Report

**Ionic Info**
```
Ionic:

   ionic (Ionic CLI)          : 4.1.1 (~/.nvm/versions/node/v8.11.1/lib/node_modules/ionic)
   Ionic Framework            : @ionic/angular 4.0.0-beta.5
   @angular-devkit/core       : 0.7.5
   @angular-devkit/schematics : 0.7.5
   @angular/cli               : 6.1.5
   @ionic/ng-toolkit          : 1.0.7
   @ionic/schematics-angular  : 1.0.5

System:

   NodeJS : v8.11.1 (~/.nvm/versions/node/v8.11.1/bin/node)
   npm    : 5.6.0
   OS     : macOS High Sierra
```

**Describe the Bug**
The left and right margins of the ion-button slots are not inverted in RTL mode.

*LTR mode:*
<img width="765" alt="capture d ecran 2018-08-28 a 16 11 32" src="https://user-images.githubusercontent.com/1470318/44732741-e1f19380-aadd-11e8-9e89-2b51cbc791fa.png">

*RTL mode:*
<img width="766" alt="capture d ecran 2018-08-28 a 16 12 14" src="https://user-images.githubusercontent.com/1470318/44732739-e158fd00-aadd-11e8-8c1c-7df5d1b80c84.png">

**Steps to Reproduce**
1. Include an icon in start or end slot in a button in an RTL app
2. Notice the inverted right / left margins

**Related Code**
Demo app on: https://github.com/abennouna/ionic4-rtl-issue

Click on the Toggle LTR/RTL modes to notice the behavior.

**Expected Behavior**
The horizontal margins should be inverted in RTL mode.

**Additional Context**
In LTR mode, we have the following generated CSS:
```css
ion-icon[slot=start] {
  margin: 0 0.3em 0 -0.3em;
}
````

In RTL mode, it should be:
```css
ion-icon[slot=start] {
  margin: 0 -0.3em 0 0.3em;
}
````

**Comment by abennouna** [comment]
Demo: https://abennouna.github.io/home

**Comment by abennouna** [comment]
Yes: RTL mode is broken out of the box, and it's not clear how we can make it work with static CSS files.

As a workaround, we [use SCSS files for the moment](https://github.com/ionic-team/ionic/issues/15341#issuecomment-417674634).

You can check the [updated demo here](https://abennouna.github.io) and the [demo repository here](https://github.com/abennouna/ionic4-rtl-issue)

**Comment by abennouna** [comment]
Closed in favor of #17012

**Comment by abennouna** [comment]
Fixed in #16987
